content/docs/19-kubernetes-knowlege-hub/00-how-to/06-deploy-stateless-frontend-app.md

```diff
@@ -0,0 +1,208 @@
+```
+
+The output of running the command above is a YAML representation of the deployment, likely similar to the lines below:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: date-suggestions
+spec:
+  selector:
+    matchLabels:
+      app: date-suggestions
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: date-suggestions
+    spec:
+      containers:
+        - name: date-suggestions
+          image: date-suggestions
+          ports:
+            - containerPort: 3000
+```
+
+ 
+modify the command to create the deployment, to add a command to pipe the output into a `deployment.yaml` file.
+
+```bash
+kubectl create deploy date-suggestions --image=date-suggestions --replicas=2 --port=3000 --dry-run=client -o yaml > deployment.yaml
+
+```
+
+Alternatively, you can use the `touch` command to create the `deployment.yaml` file, and then copy the YAML output from the command to create a deployment to it.
+
+```bash
+touch deployment.yaml
+```
+
+## Create a Kubernetes Service.
+
+Create and populate a Kubernetes Service file in the app's root directory. As the default, your application will only be accessible within the cluster. You'll need to create a Kubernetes service resource to expose it to the outside world. A service resource creates an abstraction over a set of pods that provides discovery and routing between them.
+
+To create a service, use the `kubectl expose` command as shown:
+
+```bash
+kubectl expose deployment date-suggestions --type=LoadBalancer --port=80 --target-port=3000 --name=date-suggestion-service --dry-run=client -o yaml
+```
+
+The output of running the command will be similar to the YAML below:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: date-suggestions-service
+spec:
+  type: LoadBalancer
+  selector:
+    app: date-suggestions
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 3000
+```
+
+If everything looks good, modify the command to expose the deployment, to add a command to add the output to a file `service.yaml`.
+
+```bash
+kubectl expose deployment date-suggestions --type=LoadBalancer --port=80 --target-port=3000 --name=date-suggestion-service --dry-run=client -o yaml > service.yaml
+```
+
+You can also create a YAML file with the `touch` command and add the output of the `kubectl expose` command to it.
+
+```bash
+touch service.yaml
+```
+
+Copy and paste the following line of code to the service file.
+
+## Deploy the Application.
+
+Use the `Kubectl` command line connected to the cluster you created earlier, deploy the application by applying the file's content to Kubernetes.
+
+```bash
+kubectl apply -f deployment.yaml -f service.yaml
+```
+
+## Confirm that deployment was successful.
+
+Once the deployment and service files have been applied, you should be able to access your app by running the following command:
+
+```bash
```

Comment:
While this is helpful, it could be confusing for newer users. A more user-friendly approach is to provide the user with a command that gives them exactly what they are looking for. Below is the command I would have provided the user.

```
kubectl get service date-suggestions-service --output=jsonpath='{.status.loadBalancer.ingress[0].ip}'

```